metrics/config.go

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"contrib.go.opencensus.io/exporter/stackdriver"
+	"go.opencensus.io/exporter/prometheus"
+	"go.opencensus.io/stats/view"
+	"go.uber.org/zap"
+	monitoredrespb "google.golang.org/genproto/googleapis/api/monitoredres"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	ObservabilityConfigName = "config-observability"
+	backendDestinationKey   = "metrics.backend-destination"
+	stackdriverProjectIdKey = "metrics.stackdriver-project-id"
```

Comment:
Capitalize `d` (see below).

Do we expect there to be cases where this is something other than the cluster's project?

If not, consider using: https://godoc.org/cloud.google.com/go/compute/metadata#ProjectID

Reply:
yes, this is the stackdriver project id, not necessarily the GCP project id.